AirCasting/CoreData/SessionStorage.swift

```diff
@@ -0,0 +1,19 @@
+// Created by Lunar on 27/05/2021.
+//
+
+import Foundation
+import CoreData
+
+final class SessionStorage: ObservableObject {
+    let persistenceController: PersistenceController
+    init(persistenceController: PersistenceController) {
+        self.persistenceController = persistenceController
+    }
+
+    func clearAllSession() throws {
+        let deleteRequest = NSBatchDeleteRequest(fetchRequest: SessionEntity.fetchRequest())
+        deleteRequest.resultType = .resultTypeObjectIDs
+        let res = try persistenceController.editContext().execute(deleteRequest)
```

Comment:
This code will not inform any `NSManagedObjectContext` instance in use about the change made to the store. Take a look at the [Apple guide](https://developer.apple.com/library/archive/featuredarticles/CoreData_Batch_Guide/BatchDeletes/BatchDeletes.html) to batch delete requests. IFK if this is the expected behavior here, just pointing this CoreData "gotcha" out.

Reply:
good catch. in current usage it was intended but maybe not a good idea in global scope.